packages/runner/injection/multi-domain.js

```diff
@@ -9,16 +9,29 @@
 
 import { createTimers } from './timers'
 
-// TODO: don't hard-code the index. need it to be predictable or need
-// to search for the right one somehow. will need to be fixed when we
-// test out visiting a 3rd domain
-const cyBridgeFrame = window.parent.frames[2]
-
-const Cypress = cyBridgeFrame.Cypress
+let Cypress
+
+for (let index = 0; index < window.parent.frames.length; index++) {
+  const frame = window.parent.frames[index]
+
+  try {
+    // If Cypress is defined and we haven't gotten a cross domain error we have found the correct bridge.
+    if (frame.Cypress) {
+      Cypress = frame.Cypress
+    }
+  } catch (error) {
+    // Catch DOMException: Blocked a frame from accessing a cross-origin frame.
+    if (error.name !== 'SecurityError') {
+      throw error
+    }
+  }
+}
 
+// TODO: If the spec bridge is not found we should throw some kind of error to main cypress, this should account for redirects so maybe wait a bit before throwing?
+// This may not be needed if we defer to the first command
 if (!Cypress) {
   throw new Error('Something went terribly wrong and we cannot proceed. We expected to find the global \
-    Cypress in the spec bridge window but it is missing. This should never happen and likely is a bug. Please open an issue.')
+    Cypress in the spec bridge window but it is missing.')
```

Comment:
This could legitimately happen if a click causes several redirects and finally lands on the specified domain. That shouldn't be an error.

Reply:
So do we even want this error? Seems odd to say something went terribly wrong for an expected scenario.

Reply:
I think we will want to remove this error in the long run. Not finding cypress may or may not be a problem depending if this is the last domain loaded and I don't think we can know that. Even if it is the last domain loaded this error isn't bubbled up to the primary domain to display and the test times out.

I think the onus lies on the primary domain and ideally, the command that interacts with the dom to timeout if the correct domain hasn't loaded in an appropriate amount of time.

Reply:
+1, I somehow missed the comment you added above the line